src/FormatNumber/Parser.elm

```diff
@@ -378,3 +380,40 @@ parse locale original =
                 | prefix = locale.zeroPrefix
                 , suffix = locale.zeroSuffix
             }
+
+{-| Given a `Locale` parses a `String` into a `Maybe Float`:
+-}
+parseString : Locale -> String -> Maybe Float
+parseString locale value = 
+    let
+        isNegative : Bool
+        isNegative = 
+            if String.left 1 value == locale.negativePrefix then
```

Comment:
This is probably not the right approach because:

1. `Locale.negativePrefix` is a string and it is not guaranteed to be 1 character long, one could have, for example `" -"`, for example; the numbers `"-123"` and `" -123"` would both be incorrectly labeled as positive
2. We have  `Locale.negativeSuffix`, so one can have `""` as a prefix and `"-"` as a suffix, for example; the number `"123-` would be incorrectly interpreted as a positive number

Reply:
We can use the following regex pattern to match negative number by using both prefix and suffix. But, we will run into problem when Prefix or, suffix are brackets (either ( or, [) as they have special regex meaning. We would have to check for them and handle them appropriately.

`^locale.negativePrefix.*locale.negativeSuffix$`

Reply:
Updated this using `String.startsWith` and `String.endsWith` to match prefix and suffix.